src/actions/phoneContactsActions.js

```diff
@@ -0,0 +1,166 @@
+  type: PHONE_CONTACTS_RECEIVED,
+  payload: contacts,
+});
+
+const phoneContactsError = (): PhoneContactsErrorAction => ({
+  type: PHONE_CONTACTS_ERROR,
+});
+
+const fetchingPhoneContacts = (): FetchingPhoneContactsAction => ({
+  type: FETCHING_PHONE_CONTACTS,
+});
+
+const filterDuplicateEmails = (emailAddresses: PhoneContactEmail[]): PhoneContactEmail[] => {
+  return emailAddresses.reduce((uniqueEmails, emailItem) => {
+    if (!uniqueEmails.some(({ email }) => email === emailItem.email)) {
+      return [...uniqueEmails, { ...emailItem }];
+    }
+
+    return uniqueEmails;
+  }, []);
+};
+
+const filterDuplicatePhones = (phoneNumbers: PhoneContactPhone[]): PhoneContactPhone[] => {
+  return phoneNumbers.reduce((uniqueValidPhones, phoneItem) => {
+    const phoneWithoutSpaces = stringWithoutSpaces(phoneItem.number);
+
+    if (!uniqueValidPhones.some(({ number }) => number === phoneWithoutSpaces)) {
+      return [...uniqueValidPhones, { ...phoneItem, number: phoneWithoutSpaces }];
+    }
+
+    return uniqueValidPhones;
+  }, []);
+};
+
+const formatContacts = (contacts: PhoneContact[]): ReferralContact[] => {
+  return contacts.reduce((array, contact) => {
+    const {
+      recordID,
+      displayName,
+      emailAddresses,
+      phoneNumbers,
+      thumbnailPath,
+    } = contact;
+    const formattedContact = {
+      name: displayName,
+      photo: thumbnailPath,
+    };
+    const arrayOfContacts = [];
+
+    if (emailAddresses.length) {
+      filterDuplicateEmails(emailAddresses)
+        .forEach((email) => {
+          arrayOfContacts.push({
+            ...formattedContact,
+            id: `${recordID}-${email.id}`,
+            email: email.email,
+          });
+        });
+    }
+
+    if (phoneNumbers.length) {
+      filterDuplicatePhones(phoneNumbers)
+        .forEach((phone) => {
+          arrayOfContacts.push({
+            ...formattedContact,
+            id: `${recordID}-${phone.id}`,
+            phone: phone.number,
+          });
+        });
+    }
+
+    if (arrayOfContacts.length) {
+      return [...array, ...arrayOfContacts];
+    }
+    return array;
+  }, []);
+};
+
+const sortContacts = (contacts: ReferralContact[]): ReferralContact[] => {
+  return contacts.sort((a, b) => {
+    if (a.name > b.name) return 1;
+    if (a.name < b.name) return -1;
+    return 0;
+  });
+};
+
+const askPermission = () => {
+  if (Platform.OS === 'ios') {
+    return Promise.resolve();
+  }
+
+  return PermissionsAndroid.request(PermissionsAndroid.PERMISSIONS.READ_CONTACTS, {
+    title: 'Contacts',
+    message: 'This app would like to view your contacts.',
+    buttonPositive: 'Allow',
+  });
+};
+
+export const fetchPhoneContactsAction = () => {
+  return async (dispatch: Dispatch) => {
+    askPermission()
+      .then(() => {
+        dispatch(fetchingPhoneContacts());
+        Contacts.getAll((err, contacts) => {
+          if (err === 'denied') {
```

Comment:
Just to make it more stable I would:
1. Make sure err is string and lowercase it.
2. Try to import constant from library that stands for `denied` error.

Reply:
Updated.